### PR TITLE
Allows multiple columns in domain estimation

### PR DESCRIPTION
**Pull request by smishr**
Allows estimators to take any specification of columns which `groupby` accepts.

### CODE REVIEW

**Comment by ayushpatnaikgit** [comment]
Perhaps add a test case where `by` is `[:cname, :stype]`?

**Comment by smishr** [comment]
@asinghvi17 do you want to add anything else for this PR?

**Comment by asinghvi17** [comment]
No, just some tests.  Will get those latest by Monday.

**Comment by asinghvi17** [comment]
I can't commit to this branch, but since I removed all constraints on the index type, we could copy-paste some of the current tests, with the column specifier being `Cols(==(colname))` instead of whatever `colname` was there before.  This should test that the passthrough is successful.

**Comment by asinghvi17** [comment]
I can't commit to this branch, but since I removed all constraints on the index type, we could copy-paste some of the current tests, with the column specifier being `Cols(==(colname))` instead of whatever `colname` was there before.  This should test that the passthrough is successful.

**Comment by smishr** [comment]
> I can't commit to this branch

I have added you with write permissions for the repository. You should be able to now.

> copy-paste some of the current tests, with the column specifier being `Cols(==(colname))` instead of whatever `colname` was there before.  This should test that the passthrough is successful.

This sounds good. @sayantikaSSG @itsdebartha may help in this

File: test/total.jl [hunk 1]
```diff
@@ -168,4 +168,13 @@ end
     # equivalent R code (results cause clutter):
     # > svyby(~api00, ~cname, clus1rep, svytotal)
     # > svyby(~api00, ~cname, clus1rep, svymean)
+    
+    # Test that column specifiers from DataFrames make it through this pipeline
+    # These tests replicate what you see above...just with a different syntax.
+    tot = total(:api00, Cols(==(:cname)), dclus1)
```
Comment (asinghvi17) [inline test/total.jl:174]:
```suggestion
    tot = total(:api00, Survey.DataFrames.Cols(==(:cname)), dclus1)

**Review by smishr — COMMENTED** [review]
current tests are failing. lets try to push through and merge this PR asap @asinghvi17

**Comment by asinghvi17** [comment]
Thanks @smishr!  It should, if only because it's a good test of how general the function is.

**Comment by smishr** [comment]
> Thanks @smishr!  It should, if only because it's a good test of how general the function is.

right so ideally this (as well as other) functions in the package which currently accept symbol should work with this format as well

**Review by ayushpatnaikgit — APPROVED** [review]

**Pull request merged**